Momento/Incubating/Responses/CacheSetFetchResponse.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections.Generic;
+using System.Linq;
+using Google.Protobuf;
+using Google.Protobuf.Collections;
+using CacheClient;
+using MomentoSdk.Internal;
+using MomentoSdk.Responses;
+
+namespace MomentoSdk.Incubating.Responses;
+
+public class CacheSetFetchResponse
+{
+    public CacheGetStatus Status { get; private set; }
+    private readonly RepeatedField<ByteString>? elements;
+
+    public CacheSetFetchResponse(_SetFetchResponse response)
+    {
+        Status = (response.SetCase == _SetFetchResponse.SetOneofCase.Found) ? CacheGetStatus.HIT : CacheGetStatus.MISS;
+        elements = (Status == CacheGetStatus.HIT) ? response.Found.Elements : null;
+    }
+
+    public HashSet<byte[]>? ByteArraySet
+    {
+        get
+        {
+            if (elements == null)
+            {
+                return null;
+            }
+            return new HashSet<byte[]>(
+                elements.Select(element => element.ToByteArray()),
+                Utils.ByteArrayComparer
+            );
+        }
+    }
+
+    public HashSet<string>? StringSet()
+    {
```

Comment:
Why is this a method while ByteArraySet is a property?

Reply:
(Same question came up on the dictionary PR): because string conversion could raise an exception. I'm going to make a spec in the near future where we can agree on this in particular.